workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ViewRegistry.kt

```diff
@@ -139,10 +139,12 @@ public fun <RenderingT : Any> ViewRegistry.buildView(
     contextForNewView,
     container
   )
-    .apply {
-      check(this.getRendering<Any>() != null) {
-        "View.bindShowRendering should have been called for $this, typically by the " +
+    .also { newView ->
+      check(newView.getRendering<Any>() != null) {
+        "View.bindShowRendering should have been called for $newView, typically by the " +
           "${ViewFactory::class.java.name} that created it."
+
+        WorkflowLifecycleOwner.installOn(newView)
```

Comment:
I ended up actually not using the view initializer thing for this. I haven't tested this, so i'm not sure if it actually works, but I suspect that it's reasonable to require anyone trying to use the view tree stuff to wait until the view is attached (which should happen long after both the initial showRendering and this call). The compose infra does exactly that – it doesn't validate the view tree owners until it's attached.